pentasharp/Filters/AdminOnlyFilter.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace WebApplication1.Filters
+{
+    public class AdminOnlyFilter : ActionFilterAttribute
+    {
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            var isAdmin = context.HttpContext.Session.GetString("IsAdmin") == "true";
```

Comment:
we should stop accessing HttpContext directly, use dependency injection here as well